### DISCUSSION TITLE
About heat flux during heat conduction

### DISCUSSION THREAD

**Discussion by abc-k** [firemodels/fds#12349] [Q&A]
I am trying to investigate heat transfer, including convection and radiation, to a door near a heat source.
Therefore, I made some modifications to the sample file (back_wall_test.fds) to output the temperature of the door.
The modifications at this point are as follows.
•Initialized the EMISSIVITY of the steel (EMISSIVITY=1 -> EMISSIVITY=0.9).
•Removed  ”HEAT_TRANSFER_COEFFICIENT=0” from STEEL PLATE and HOT.
•Temperature output devices were added to the wall surface.
•Temperature SLCFs were added at each axis origin.

Comparing the results of this modified sample file (A.fds) and the two cases (g981.fds,EMISSIVITY0.fds) created based on the modified sample file (A.fds), I found the following unknowns in each.

(1) Initialization of GVEC (changing g from 0 to 9.81). (file:g981.fds)
I want to consider combustion in the upper layer, so I want to make gravity work and create a high temperature layer. So I removed "GVEC=0,0,0" and initialized it.
From Equation 8.3 in the user guide, if g=9.81, h should be higher than when g=0. Therefore, I thought that more heat would be transferred to the outside if GVEC was initialized.
However, as shown in the figure below, the result is that less heat is transferred to the outside than when g=0. 

![image](https://github.com/firemodels/fds/assets/150226734/25683191-e33e-43c8-80af-4c3b31151de8)

Q1: Could you tell me what caused the loss of heat transfer when g was changed from 0 to 9.81?

(2) I set EMISSIVITY=0 and tried to see if only the convection term could be output. (file:EMISSIVITY0.fds)
"back_wall_test.fds" outputs only the radiative heat flux term. I tried to see if it is possible to output only the convective heat flux term.
From Equation 22.33 in the User's Guide, I thought that if EMISSIVITY = 0, then εs = 0 and only the radiative heat flux term could be set to 0. The result is shown in the graph below. Where QC_O is the convective heat flux incident on the heater-side wall, QC_I is the convective heat flux incident on the opposite wall, QR_O is the radiant heat flux incident on the heater-side wall, and QR_I is the radiant heat flux incident on the opposite wall.

![image](https://github.com/firemodels/fds/assets/150226734/6d181669-a77a-45f1-a8bc-442820e39c97)

In fact, when I set EMISSIVITY=0 and output the results, I confirmed that RADIATIVE_HEAT_FLUX is set to 0. On the other hand, the value of CONVECTIVE_HEAT_FLUX was different from that when EMISSIVITY=0.9.

Q2:Does the value of EMISSIVITY also affect the convective heat flux term?
Q3: Could you please tell me why the value of CONVECTIVE_HEAT_FLUX was different for EMISSIVITY=0 and EMISSIVITY=0.9?

I appreciate your clarification on the above three questions.

**Comment by abc-k** [comment]
Create a steel box with 1 m sides. Add a hot inner box 0.8 m on a side centered inside the box, and then record 
external temperatures of the box sides that are zero or one cell thick. It should not matter whether the sides are zero
or one cell thick -- the heat conduction should be the same because the steel is 0.5 cm thick on all sides.
Note that the sides of the box coincide with mesh boundaries. This tests the ability of the code to transfer heat across mesh
boundaries within a solid. For simplicity there is no convective heat transfer, emissivities are all 1, and the background
gas is nitrogen. The heat transfer is purely radiative in the gas phase and conductive through the solid walls.

&HEAD CHID='A', TITLE='Check heat transfer through interpolated boundaries' /

&MESH IJK=21,20,20, XB=-0.50, 0.55,-0.5, 0.5,-0.5, 0.5 /
&MESH IJK=19,20,20, XB= 0.55, 1.50,-0.5, 0.5,-0.5, 0.5 /
&MESH IJK=20,20,20, XB=-1.50,-0.50,-0.5, 0.5,-0.5, 0.5 /
&MESH IJK=20,20,20, XB=-0.50, 0.50, 0.5, 1.5,-0.5, 0.5 /

&MISC GVEC=0.,0.,0., NOISE=.FALSE., STRATIFICATION=.FALSE. /

&TIME T_END=3600., DT=0.5 /

&SPEC ID='NITROGEN', BACKGROUND=.TRUE. /

&MATL ID                 = 'STEEL'
      SPECIFIC_HEAT      = 0.5
      CONDUCTIVITY       = 50.
      DENSITY            = 8000. /!EMISSIVITY         = 1.

&SURF ID='HOT', TMP_FRONT=500., EMISSIVITY=1., COLOR='RED' /

&SURF ID        = 'STEEL PLATE'
      DEFAULT   = .TRUE.
      COLOR     = 'BLACK'
      MATL_ID   = 'STEEL'
      THICKNESS = 0.005/
      
&OBST XB=-0.40, 0.40,-0.40, 0.40,-0.40, 0.40, SURF_ID='HOT' / 

&OBST XB=-0.55,-0.50,-0.50, 0.50,-0.50, 0.50 /
&OBST XB= 0.50, 0.55,-0.50, 0.50,-0.50, 0.50 /
&OBST XB=-0.50, 0.50,-0.55,-0.50,-0.50, 0.50 /
&OBST XB=-0.50, 0.50, 0.50, 0.50,-0.50, 0.50 /
&OBST XB=-0.50, 0.50,-0.50, 0.50,-0.50,-0.50 /
&OBST XB=-0.50, 0.50,-0.50, 0.50, 0.50, 0.50 /

&VENT MB='XMIN', SURF_ID='OPEN' /
&VENT MB='XMAX', SURF_ID='OPEN' /
&VENT MB='YMIN', SURF_ID='OPEN' /
&VENT MB='YMAX', SURF_ID='OPEN' /
&VENT MB='ZMIN', SURF_ID='OPEN' /
&VENT MB='ZMAX', SURF_ID='OPEN' /

&BNDF QUANTITY='WALL TEMPERATURE', CELL_CENTERED=.TRUE. /

&DEVC XYZ=0.49,0.00,0.00, QUANTITY='RADIATIVE HEAT FLUX', ID='Flux_in', IOR=-1 /
&DEVC XYZ=0.56,0.00,0.00, QUANTITY='RADIATIVE HEAT FLUX', ID='Flux_out', IOR=1, CONVERSION_FACTOR=-1. /

&DEVC XYZ=-.56,0.00,0.00, QUANTITY='WALL TEMPERATURE', ID='Temp_s_o', IOR=-1 /
&DEVC XYZ=-.49,0.00,0.00, QUANTITY='WALL TEMPERATURE', ID='Temp_s_i', IOR=1 /
&DEVC XYZ=-.6,0.00,0.00, QUANTITY='TEMPERATURE', ID='Temp_g_o'/
&DEVC XYZ=-.45,0.00,0.00, QUANTITY='TEMPERATURE', ID='Temp_g_i'/
&DEVC XYZ=0.00,0.51,0.00, QUANTITY='WALL TEMPERATURE', ID='Temp_thin', IOR=2 /

&SLCF PBY=0., QUANTITY='INTEGRATED INTENSITY', CELL_CENTERED=.TRUE. /
&SLCF PBY=0., QUANTITY='TEMPERATURE' /
&SLCF PBX=0., QUANTITY='TEMPERATURE' /
&SLCF PBZ=0., QUANTITY='TEMPERATURE' /

&DEVC ID='QC_O',XYZ=-.56,0.00,0.00,QUANTITY='CONVECTIVE HEAT FLUX',IOR=-1/
&DEVC ID='QR_O',XYZ=-.56,0.00,0.00,QUANTITY='RADIATIVE HEAT FLUX',IOR=-1/
&DEVC ID='QN_O',XYZ=-.56,0.00,0.00,QUANTITY='NET HEAT FLUX',IOR=-1/

&DEVC ID='QC_I',XYZ=-.49,0.00,0.00,QUANTITY='CONVECTIVE HEAT FLUX',IOR=1/
&DEVC ID='QR_I',XYZ=-.49,0.00,0.00,QUANTITY='RADIATIVE HEAT FLUX',IOR=1/
&DEVC ID='QN_I',XYZ=-.49,0.00,0.00,QUANTITY='NET HEAT FLUX',IOR=1/



&TAIL /

**Reply by abc-k** [reply]
Create a steel box with 1 m sides. Add a hot inner box 0.8 m on a side centered inside the box, and then record 
external temperatures of the box sides that are zero or one cell thick. It should not matter whether the sides are zero
or one cell thick -- the heat conduction should be the same because the steel is 0.5 cm thick on all sides.
Note that the sides of the box coincide with mesh boundaries. This tests the ability of the code to transfer heat across mesh
boundaries within a solid. For simplicity there is no convective heat transfer, emissivities are all 1, and the background
gas is nitrogen. The heat transfer is purely radiative in the gas phase and conductive through the solid walls.

&HEAD CHID='EMISSIVITY0', TITLE='Check heat transfer through interpolated boundaries' /

&MESH IJK=21,20,20, XB=-0.50, 0.55,-0.5, 0.5,-0.5, 0.5 /
&MESH IJK=19,20,20, XB= 0.55, 1.50,-0.5, 0.5,-0.5, 0.5 /
&MESH IJK=20,20,20, XB=-1.50,-0.50,-0.5, 0.5,-0.5, 0.5 /
&MESH IJK=20,20,20, XB=-0.50, 0.50, 0.5, 1.5,-0.5, 0.5 /

&MISC GVEC=0.,0.,0., NOISE=.FALSE., STRATIFICATION=.FALSE. /

&TIME T_END=3600., DT=0.5 /

&SPEC ID='NITROGEN', BACKGROUND=.TRUE. /

&MATL ID                 = 'STEEL'
      EMISSIVITY         = 0
      SPECIFIC_HEAT      = 0.5
      CONDUCTIVITY       = 50.
      DENSITY            = 8000. /

&SURF ID='HOT', TMP_FRONT=500., EMISSIVITY=1., COLOR='RED' /

&SURF ID        = 'STEEL PLATE'
      DEFAULT   = .TRUE.
      COLOR     = 'BLACK'
      MATL_ID   = 'STEEL'
      THICKNESS = 0.005
      TRANSPARENCY=0.8 /
      
&OBST XB=-0.40, 0.40,-0.40, 0.40,-0.40, 0.40, SURF_ID='HOT' / 

&OBST XB=-0.55,-0.50,-0.50, 0.50,-0.50, 0.50 / 
&OBST XB= 0.50, 0.55,-0.50, 0.50,-0.50, 0.50 / 
&OBST XB=-0.50, 0.50,-0.55,-0.50,-0.50, 0.50 / 
&OBST XB=-0.50, 0.50, 0.50, 0.50,-0.50, 0.50 / 
&OBST XB=-0.50, 0.50,-0.50, 0.50,-0.50,-0.50 / 
&OBST XB=-0.50, 0.50,-0.50, 0.50, 0.50, 0.50 / 

&VENT MB='XMIN', SURF_ID='OPEN' /
&VENT MB='XMAX', SURF_ID='OPEN' /
&VENT MB='YMIN', SURF_ID='OPEN' /
&VENT MB='YMAX', SURF_ID='OPEN' /
&VENT MB='ZMIN', SURF_ID='OPEN' /
&VENT MB='ZMAX', SURF_ID='OPEN' /

&BNDF QUANTITY='WALL TEMPERATURE', CELL_CENTERED=.TRUE. /

&DEVC XYZ=0.49,0.00,0.00, QUANTITY='RADIATIVE HEAT FLUX', ID='Flux_in', IOR=-1 /
&DEVC XYZ=0.56,0.00,0.00, QUANTITY='RADIATIVE HEAT FLUX', ID='Flux_out', IOR=1, CONVERSION_FACTOR=-1. /

&DEVC XYZ=-.56,0.00,0.00, QUANTITY='WALL TEMPERATURE', ID='Temp_s_o', IOR=-1 /
&DEVC XYZ=-.49,0.00,0.00, QUANTITY='WALL TEMPERATURE', ID='Temp_s_i', IOR=1 /
&DEVC XYZ=-.6,0.00,0.00, QUANTITY='TEMPERATURE', ID='Temp_g_o'/
&DEVC XYZ=-.45,0.00,0.00, QUANTITY='TEMPERATURE', ID='Temp_g_i'/
&DEVC XYZ=0.00,0.51,0.00, QUANTITY='WALL TEMPERATURE', ID='Temp_thin', IOR=2 /

&SLCF PBY=0., QUANTITY='INTEGRATED INTENSITY', CELL_CENTERED=.TRUE. /
&SLCF PBY=0., QUANTITY='TEMPERATURE' /
&SLCF PBX=0., QUANTITY='TEMPERATURE' /
&SLCF PBZ=0., QUANTITY='TEMPERATURE' /

&DEVC ID='QC_O',XYZ=-.56,0.00,0.00,QUANTITY='CONVECTIVE HEAT FLUX',IOR=-1/
&DEVC ID='QR_O',XYZ=-.56,0.00,0.00,QUANTITY='RADIATIVE HEAT FLUX',IOR=-1/
&DEVC ID='QN_O',XYZ=-.56,0.00,0.00,QUANTITY='NET HEAT FLUX',IOR=-1/

&DEVC ID='QC_I',XYZ=-.49,0.00,0.00,QUANTITY='CONVECTIVE HEAT FLUX',IOR=1/
&DEVC ID='QR_I',XYZ=-.49,0.00,0.00,QUANTITY='RADIATIVE HEAT FLUX',IOR=1/
&DEVC ID='QN_I',XYZ=-.49,0.00,0.00,QUANTITY='NET HEAT FLUX',IOR=1/

&TAIL /

**Reply by abc-k** [reply]
Create a steel box with 1 m sides. Add a hot inner box 0.8 m on a side centered inside the box, and then record 
external temperatures of the box sides that are zero or one cell thick. It should not matter whether the sides are zero
or one cell thick -- the heat conduction should be the same because the steel is 0.5 cm thick on all sides.
Note that the sides of the box coincide with mesh boundaries. This tests the ability of the code to transfer heat across mesh
boundaries within a solid. For simplicity there is no convective heat transfer, emissivities are all 1, and the background
gas is nitrogen. The heat transfer is purely radiative in the gas phase and conductive through the solid walls.

&HEAD CHID='g981', TITLE='Check heat transfer through interpolated boundaries' /

&MESH IJK=21,20,20, XB=-0.50, 0.55,-0.5, 0.5,-0.5, 0.5 /
&MESH IJK=19,20,20, XB= 0.55, 1.50,-0.5, 0.5,-0.5, 0.5 /
&MESH IJK=20,20,20, XB=-1.50,-0.50,-0.5, 0.5,-0.5, 0.5 /
&MESH IJK=20,20,20, XB=-0.50, 0.50, 0.5, 1.5,-0.5, 0.5 /

&MISC  NOISE=.FALSE., STRATIFICATION=.FALSE. /

&TIME T_END=3600., DT=0.5 /

&SPEC ID='NITROGEN', BACKGROUND=.TRUE. /

&MATL ID                 = 'STEEL'
      SPECIFIC_HEAT      = 0.5
      CONDUCTIVITY       = 50.
      DENSITY            = 8000. /!EMISSIVITY         = 1.

&SURF ID='HOT', TMP_FRONT=500., EMISSIVITY=1., COLOR='RED' /

&SURF ID        = 'STEEL PLATE'
      DEFAULT   = .TRUE.
      COLOR     = 'BLACK'
      MATL_ID   = 'STEEL'
      THICKNESS = 0.005
      TRANSPARENCY=0.8 /!HEAT_TRANSFER_COEFFICIENT=0.
      
&OBST XB=-0.40, 0.40,-0.40, 0.40,-0.40, 0.40, SURF_ID='HOT' / 

&OBST XB=-0.55,-0.50,-0.50, 0.50,-0.50, 0.50 / 
&OBST XB= 0.50, 0.55,-0.50, 0.50,-0.50, 0.50 / 
&OBST XB=-0.50, 0.50,-0.55,-0.50,-0.50, 0.50 / 
&OBST XB=-0.50, 0.50, 0.50, 0.50,-0.50, 0.50 / 
&OBST XB=-0.50, 0.50,-0.50, 0.50,-0.50,-0.50 / 
&OBST XB=-0.50, 0.50,-0.50, 0.50, 0.50, 0.50 / 

&VENT MB='XMIN', SURF_ID='OPEN' /
&VENT MB='XMAX', SURF_ID='OPEN' /
&VENT MB='YMIN', SURF_ID='OPEN' /
&VENT MB='YMAX', SURF_ID='OPEN' /
&VENT MB='ZMIN', SURF_ID='OPEN' /
&VENT MB='ZMAX', SURF_ID='OPEN' /

&BNDF QUANTITY='WALL TEMPERATURE', CELL_CENTERED=.TRUE. /

&DEVC XYZ=0.49,0.00,0.00, QUANTITY='RADIATIVE HEAT FLUX', ID='Flux_in', IOR=-1 /
&DEVC XYZ=0.56,0.00,0.00, QUANTITY='RADIATIVE HEAT FLUX', ID='Flux_out', IOR=1, CONVERSION_FACTOR=-1. /

&DEVC XYZ=-.56,0.00,0.00, QUANTITY='WALL TEMPERATURE', ID='Temp_s_o', IOR=-1 /
&DEVC XYZ=-.49,0.00,0.00, QUANTITY='WALL TEMPERATURE', ID='Temp_s_i', IOR=1 /
&DEVC XYZ=-.6,0.00,0.00, QUANTITY='TEMPERATURE', ID='Temp_g_o'/
&DEVC XYZ=-.45,0.00,0.00, QUANTITY='TEMPERATURE', ID='Temp_g_i'/
&DEVC XYZ=0.00,0.51,0.00, QUANTITY='WALL TEMPERATURE', ID='Temp_thin', IOR=2 /

&SLCF PBY=0., QUANTITY='INTEGRATED INTENSITY', CELL_CENTERED=.TRUE. /
&SLCF PBY=0., QUANTITY='TEMPERATURE' /
&SLCF PBX=0., QUANTITY='TEMPERATURE' /
&SLCF PBZ=0., QUANTITY='TEMPERATURE' /

&TAIL /

**Comment by mcgratta** [comment]
This is just too much. It is difficult to untangle all this. If you believe that the code has a bug, submit a simple case to the Issue Tracker. No one has the time to sort out all these cases and answer all your questions.
